portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,18 +27,41 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  public ArrayList<String> comments = new ArrayList<String>(List.of("boom", "bang", "pow"));
+//    public ArrayList<String> comments = new ArrayList<String>(List.of("boom", "bam", "pow"));
+
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // obtain input from form(the user's comment)
+    String userComment = getParameter(request, "userCommentInput", "");
+    
+    // seperate words into indivdual text
```

Comment:
I don't think you need to split the user comment into words. the doPost method is meant for accepting a comment from user (submitted from form). So in this doPost method, we should unpack the data and store the data in datastore.  After doing that you can simply redirect the user back to the main page, which by deafult should load you page with all comments.

So, you likely need a doGet method to handle getting the list of comments.